server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRecipientRewriteTableMailet.java

```diff
@@ -19,22 +19,28 @@
 

```

Comment:
If I follow your explanation, you should have a class that do the steps `changeReturnPath --> buildNewRecipient --> processMail --> ....` and delegate the processMail to another class.
abstract class are rarely accepable in my opinion.